### ISSUE TITLE
Consider changing the names of the native app base types (MauiWinUIApplication, MauiApplication, MauiUIApplicationDelegate)

### ISSUE THREAD

**Issue by Eilon** [dotnet/maui#2242]
These are the types in question for each platform:

* Windows (WinUI): [`MauiWinUIApplication`](https://github.com/dotnet/maui/blob/main/src/Core/src/Platform/Windows/MauiWinUIApplication.cs)
* Android: [`MauiApplication`](https://github.com/dotnet/maui/blob/main/src/Core/src/Platform/Android/MauiApplication.cs)
* iOS/MacCatalyst: [`MauiUIApplicationDelegate`](https://github.com/dotnet/maui/blob/main/src/Core/src/Platform/iOS/MauiUIApplicationDelegate.cs)

Why change the names?

- The names are not consistent with each other.
- These type names are not usually seen by developers, so using up "good" names such as `MauiApplication` might be a "waste". These types are typically used once per project in a file generated by the project template (i.e. the user doesn't type it in)
- This would mean "good" names such as `MauiApplication` could be used in more user-facing APIs that are more frequently worked on by developers, such as in the app's startup code (which, in my experience, changes quite a lot over time)

Why *not* change the names?

- The name are consistent with the native base class names, which can be a benefit for someone familiar with the native architecture.
   - This doesn't preclude doing a rename, but perhaps the rename would be an "uglier" name that bears a resemblance to the native base class.
- The names are consistent with Xamarin.Forms ([somewhat](https://github.com/xamarin/Xamarin.Forms/blob/5.0.0/Xamarin.Forms.Platform.iOS/FormsApplicationDelegate.cs))

Note: This originally came up in [this PR](https://github.com/dotnet/maui/pull/2137#issuecomment-904173606) (and subsequent discussion in the PR).

**Comment by rmarinho** [comment]
We should figure this out before going GA . Should we schedule a call ?

**Comment by Eilon** [comment]
@rmarinho yeah we could certainly do a call. I wonder does anyone have any proposals to share here, or any other thoughts on the pros/cons, or the "weights" of the pros/cons?

From my perspective, one of the biggest "weights" is to avoid "using up" a really good name such as `MauiApplication` for something that is almost never seen or interacted with directly by type name.
